src/sage/combinat/posets/bubble_and_shuffle.py

```diff
@@ -0,0 +1,203 @@
+"""
+Bubble and Shuffle lattices
+
+Shuffle lattices were defined by Greene in [Gre1988]_.
+
+Bubble lattices were introduced by McConville and Mühle in [MacCM2022]_.
+
+The Bubble lattice `B_{m,n}` and the Shuffle lattice `S_{m,n}` share the same
+underlying set.
+"""
+from typing import Iterator
+
+from sage.combinat.posets.lattices import LatticePoset
+from sage.combinat.subset import subsets
+from sage.combinat.shuffle import ShuffleProduct
+from sage.graphs.digraph import DiGraph
+from sage.graphs.graph import Graph
+from sage.rings.integer import Integer
+from sage.rings.integer_ring import ZZ
+
+
+def bubble_cardinality(m, n) -> Integer:
+    """
+    Return the cardinality of the Bubble lattice `B_{m,n}`.
+
+    This is also the cardinality of the Shuffle lattice `S_{m,n}`.
```

Comment:
```suggestion

    We have

    .. MATH::

        |B_{m,n}| = \sum_{j=0}^n \sum_{i=0}^m \binom{i+j}{j} \binom{m}{i} \binom{n}{j}.

    This is also the cardinality of the Shuffle lattice `S_{m,n}`.
```
I think it is good to be explicit about the formula in the doc here.